ERCS/erc-7555.md

```diff
@@ -0,0 +1,236 @@
+- **Provider** - A third party service provider that is able to authenticate a user and produce a keypair for the user.
+
+### Redirects
+An application looking to authenticate a user must navigate the user to a given provider's URI based on the `URI Request Syntax`. The application must implement a valid redirect URI for the callback in order to receive a valid response.
+
+#### Available Routes
+- `/auth/`: The route used to authneticate a user, and request credentials.
+- `/sendTransaction/`: The route used to send a transaction payload for a user to sign. This is more of a convenient method to allow applications to do both authentication, and plugin registration within a single redirect, instead of requiring the user to perform two redirects.
+
+### Schema
+The signer type represents the specific curve used when generating the keypair. The following are a non-exhaustive list of possible keypair types:
+```=
+type SIGNER_TYPE: string = "secp256k1" | "P256" | "BLS12-381";
+```
+The `signer_key` should be returned in the did:key format, as specified by the W3C.
+The `smart_account_address` should be returned in the CAIP-10 format.
+
+#### Auth Route
+##### Request Schema
+```=
+ parameters:
+    - in: query
+      name: redirect_uri
+      schema:
+        type: string
+      description: The uri that the provider should redirect back to.
+   - in: query
+      name: chain_id
+      schema:
+        type: string
+      description: The chain_id of a given network.
+```
+##### Response Schema
+```=
+ parameters:
+    - in: query
+      name: signer_key
+      schema:
+        type: string
+      description: The public key of the user, formatted using W3C definition of did:key
+  - in: query
+      name: smart_account_address
+      schema:
+        type: string
+      description: The on-chain address for a given smart account, formatted using CAIP-10
+```
+
+##### Request Syntax
+```=
+https://<PROVIDER_URI>/auth/?
+    redirect_uri=<YOUR_REDIRECT_URI>
+    &chain_id=<CHAIN_ID>
+```
+##### Response Syntax
+```=
+https://<YOUR_REDIRECT_URI>/auth/?
+    signer_key=<SIGNER_KEY>
+    &smart_account_address=<SMART_ACCOUNT_ADDRESS>
+```
+
+#### sendTransaction Route
+##### Request Schema
+```=
+ parameters:
+    - in: query
+      name: redirect_uri
+      schema:
+        type: string
+      description: The uri that the provider should redirect back to.
+   - in: query
+      name: chain_id
+      schema:
+        type: string
+      description: The chain_id of a given network.
+   - in: query
+      name: transaction
+      schema:
+        type: string
+      description: The RLP encoded transaction that needs to be signed
+```
+##### Response Schema
+```=
+ parameters:
+    - in: query
+      name: signer_key
+      schema:
+        type: string
+      description: The public key of the user, formatted using W3C definition of did:key
+  - in: query
+      name: smart_account_address
```

Comment:
do we need this?

Seems like you could use this endpoint to submit tx not related to the smart contract account which has quite a neat usecase with Web3Auth where you could use the same EOA across different apps if you want.

In any case dapp already knows this from /auth endpoint

Reply:
you are correct! This was probably a bad copy/paste!

Reply:
@GregTheGreek I don't think this is resolved

Reply:
yeah still here @GregTheGreek